lpr.Logic/Services/JWTService.cs

```diff
@@ -0,0 +1,94 @@
+﻿using lpr.Common.Interfaces.Data;
+using lpr.Common.Interfaces.Services;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace lpr.Logic.Services
+{
+    public class JWTService : IJWTService
+    {
+        public string SecretKey { get; set; }
+        public JWTService(string secretKey)
+        {
+            SecretKey = secretKey;
+        }
+
+        public string GenerateToken(IJWTContainerModel model)   
+        {
+            if (model == null || model.Claims == null || model.Claims.Length == 0)
+                throw new ArgumentException("Arguments to create token are not valid");
+
+            SecurityTokenDescriptor securityTokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(model.Claims),
+                Expires = DateTime.UtcNow.AddMinutes(Convert.ToInt32(model.ExpireMinutes)),
+                SigningCredentials = new SigningCredentials(GetSymmertricSecurityKey(), model.SecurityAlogrithm)
+            };
+
+            JwtSecurityTokenHandler jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
+            SecurityToken securityToken = jwtSecurityTokenHandler.CreateToken(securityTokenDescriptor);
+
+            return jwtSecurityTokenHandler.WriteToken(securityToken);
+        }
+
+        public IEnumerable<Claim> GetTokenClaims(string token)
+        {
+            if (string.IsNullOrEmpty(token))
+                throw new ArgumentException("Given token is null or empty");
+
+            TokenValidationParameters tokenValidationParameters = GetTokenValidationParameters();
+
+            JwtSecurityTokenHandler jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
+            try
+            {
+                ClaimsPrincipal tokenValid = jwtSecurityTokenHandler.ValidateToken(token, tokenValidationParameters, out SecurityToken validatedToken);
+                return tokenValid.Claims;
+            }
+            catch (Exception e)
+            {
+                throw;
+            }
+        }
+
+        public bool IsTokenValid(string token)
+        {
+            if (string.IsNullOrEmpty(token))
+                throw new ArgumentException("Given token is null or empty");
+
+            TokenValidationParameters tokenValidationParameters = GetTokenValidationParameters();
+
+            JwtSecurityTokenHandler jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
+            try
+            {
+                ClaimsPrincipal tokenValid = jwtSecurityTokenHandler.ValidateToken(token, tokenValidationParameters, out SecurityToken validatedToken);
+                return true;
+            }
+            catch (Exception)
+            {
+                return false;
```

Comment:
Dont you need to log/print all the exceptions in console?

Reply:
We don't have a 'official' method of logging errors yet.
So we might look into this next sprint.